deluge/ui/gtkui/connectionmanager.py

```diff
@@ -233,6 +234,23 @@ def add_host(self, host, port, username='', password=''):
         # Update the status of the hosts
         self.__update_list()
 
+    def on_entry_host_paste_clipboard(self, widget):
+        paste = gtk.clipboard_get().wait_for_text()
+        log.debug('on_entry_proxy_host_paste-clipboard: got paste: %s', paste)
+        if '//' not in paste:
+            parsed = urlparse('//' + paste)
+        else:
+            parsed = urlparse(paste)
+        if parsed.hostname is not None:
+            widget.set_text(parsed.hostname)
+            widget.emit_stop_by_name('paste-clipboard')
```

Comment:
If you mean the `emit_stop_by_name`, it is required strictly when we modify the entry field manually (to stop the clipboard contents from getting pasted at the end of the text we've set).